### ISSUE TITLE
Revise how `SitesData.*_parts` data are being defined

### ISSUE THREAD

**Issue by stefsmeets** [GEMDAT-repos/GEMDAT#34]
To calculate statistics, we are calculating dedicated *_parts attributes on `SitesData`. These inputs for these analyses are no different than the full data set, just on a smaller subset.

So, instead of having dedicated *parts* variables, like `$VARIABLE_parts`, consider feeding `SitesData` with the parent data already split (probably `atom_sites` or `all_transitions`. Use these as a basis for subsequent calculations.

Basically, instead of:

`parent_data -> SitesData -> derived_data -> split -> derived_data_parts`

One full SitesData instance for entire timeseries with attributes containing lists of parts data.

Do:

```
parent_data -> SitesData -> derived_data
parent_data -> split -> SitesData -> derived_data_parts
```
One SitesData instance for entire timeseries + list of SitesData instances (one for each part).

**Comment by stefsmeets** [comment]
We could have a `SitesData.split(n_parts=10)` method:

```python
>>> sites = SitesData(structure=structure)
>>> sites.calculate_all(data=data, extras=extras)
>>> sites_parts = sites.split(n_parts=10)
>>> len(sites_parts)
10
>>> type(sites_parts)
list
>>> [type(part) for part in sites_parts]
[SitesData, SitesData, ..., SitesData]
